GithubRepo/Scenes/Search/ViewModel/SearchViewModel.swift

```diff
@@ -11,44 +11,44 @@ class SearchViewModel: ViewModelSearching {
     
     // MARK: - Attributes
     
-    var delegate: SearchViewModelDelegate?
-    private(set) var repositoryCellViewModels: [RepositoryCellViewModel] = .init([])
-    private(set) var loading: Bool = .init(false)
-    private(set) var error: Bool = .init(false)
+    weak var delegate: SearchViewModelDelegate?
+    private var githubRepository: GithubRepository
+    private(set) var repositoryCellViewModels: [RepositoryCellViewModel]
+    private(set) var loading: Bool
+    private(set) var error: Bool
+    
+    // MARK: - Constructors
+    
+    init(repository: GithubRepository = GithubMainRepository()) {
+        self.repositoryCellViewModels = []
+        self.loading = false
+        self.error = false
+        self.githubRepository = repository
+        self.githubRepository.delegate = self
+    }
     
-
-    // MARK: - Private Methods
+    // MARK: - Methods
     
-    private func onSearchRepositoryCellUpdate(_ repo: [Repository]) {
-        self.repositoryCellViewModels = repo.map { RepositoryCellViewModel(repository: $0) }
-        self.delegate?.onChangeSearchRepository(repoCellViewModels: self.repositoryCellViewModels)
+    func fetchRepositories(query: String) {
+        githubRepository.fetchRepositories(with: query)
     }
-    
-    private func onSearchLoadingUpdate(_ isLoading: Bool) {
-        self.loading = isLoading
+}
+
+// MARK: - Notifications from Repository
+
+extension SearchViewModel: GithubRepositoryDelegate {
+    func didChangeLoading() {
+        self.loading = self.githubRepository.loading
```

Comment:
If you have applied the change that I suggested before, Here you can just use the parameter value instead of access the `loading` From the object